jaxb-plugins-parent/tests/map_init/pom.xml

```diff
@@ -0,0 +1,72 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.jvnet.jaxb</groupId>
+    <artifactId>jaxb-plugins-tests</artifactId>
+    <version>4.0.6-SNAPSHOT</version>
+  </parent>
+  <artifactId>jaxb-plugins-test-map_init</artifactId>
+  <packaging>jar</packaging>
+  <name>JAXB Tools :: JAXB Plugins :: Test [map_init]</name>
+  <dependencies>
+    <dependency>
+      <groupId>org.jvnet.jaxb</groupId>
+      <artifactId>jaxb-maven-plugin-testing</artifactId>
+      <scope>test</scope>
+    </dependency>
+  </dependencies>
+  <build>
+    <defaultGoal>test</defaultGoal>
+    <plugins>
+      <plugin>
+        <groupId>org.jvnet.jaxb</groupId>
+        <artifactId>jaxb-maven-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>generate</id>
+            <configuration>
+              <extension>true</extension>
+              <schemaIncludes>
+                <include>**/example-undefinedMapClass.xsd</include>
+              </schemaIncludes>
+              <args>
+                <arg>-XMapInit</arg>
+              </args>
+              <plugins>
+                <plugin>
+                  <groupId>org.jvnet.jaxb</groupId>
+                  <artifactId>jaxb-plugins</artifactId>
+                </plugin>
+              </plugins>
+            </configuration>
+          </execution>
+          <execution>
+            <id>generate-defined</id>
+            <goals>
+              <goal>generate</goal>
+            </goals>
+            <configuration>
+              <extension>true</extension>
+              <schemaIncludes>
+                <include>**/example-definedMapClass.xsd</include>
+              </schemaIncludes>
+              <generateDirectory>${project.build.directory}/generated-sources/xjc-defined</generateDirectory>
+              <args>
+                <arg>-XMapInit</arg>
+                <arg>-XMapInit-mapClass=java.util.TreeMap</arg>
```

Comment:
How hard would it be to add an annotation to specify the concrete Map implementation for each instance?

For example, we have a globalBinding for collection type to go to a certain list class. It would be handy to have a global map type, and then since instance override.

Perhaps a follow-on feature enhancement ticket.

Reply:
I think since jaxb-ri / xjc doesnt handle map init, it's more plugin relates. But I can for sure add another option in annotation like the `ignore` part to override global (plugin) defined map implementation. I'll rework on this and push another change to this PR

Reply:
This is done with `<mapinit:class>java.util.WeakHashMap</mapinit:class>` local attribute annotation.

Updated PR with new follow-up commit and updated example / test